platform/darwin/src/MGLLocationManager_Private.h

```diff
@@ -0,0 +1,4 @@
+#import "MGLLocationManager.h"
+
+@interface CLLocationManager (MGLAdditions) <MGLLocationManager>
```

Comment:
This interface needs to be public. Otherwise, it would be impossible for the application to set a CLLocationManager as the map view’s location manager.

Reply:
The idea for the location manager is that people with advanced use cases implement their own LM for the rest of the world we took care with the "internal implementation". What would be the purpose for devs set the location manager with the default implementation if we already do that for them?

Reply:
There needs to be an obvious way to adjust one of CLLocationManager’s settings, whether it’s `desiredAccuracy` or `activityType`. I’m picturing a developer wanting to do something like this:

```swift
if let locationManager = mapView.locationManager as? CLLocationManager {
    locationManager.desiredAccuracy = kCLLocationAccuracyBestForNavigation
}
```

As it is, they’d have to guess that this is the case. It isn’t obvious that CLLocationManager already conforms to MGLLocationManager via that private implementation of `-authorizationStatus`.

I’m second-guessing my original idea of making CLLocationManager conform to MGLLocationManager due to the prefix issue, but I still think the developer should be able to customize just the `desiredAccuracy` or `activityType` without having to define their own, full-fledged custom location manager. This really isn’t an advanced use case. So if we go back to having the default location manager _own_ a CLLocationManager, we should make that default implementation public and possibly mirror all the options that CLLocationManager provides.

Reply:
You have a point. Nonetheless if you implement a custom LM you could control the accuracy in your own terms without requiring acknowledge by the map view. The methods in `MGLLocationManager` and `MGLLocationManagerDelegate` are the ones that are used directly in `MGLMapView`.